miner/block_builder.go

```diff
@@ -178,30 +179,62 @@ func calcHdistRange(id types.LayerID, hdist types.LayerID) (bottom types.LayerID
 	return bottom, top
 }
 
-func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.AtxId, eligibilityProof types.BlockEligibilityProof,
-	txids []types.TransactionId, atxids []types.AtxId) (*types.Block, error) {
-
+func (t *BlockBuilder) getVotes(id types.LayerID) ([]types.BlockID, error) {
 	var votes []types.BlockID = nil
-	var err error
+
+	// if genesis
 	if id == config.Genesis {
 		return nil, errors.New("cannot create blockBytes in genesis layer")
-	} else if id == config.Genesis+1 {
-		votes = append(votes, mesh.GenesisBlock.Id())
-	} else { // get from hare
-		bottom, top := calcHdistRange(id, t.hdist)
-		votes, err = t.hareResult.GetResult(bottom, top)
-		if err != nil {
-			t.With().Warning("Could not get hare result during block creation",
-				log.Uint64("bottom", uint64(bottom)), log.Uint64("top", uint64(top)),
-				log.Uint64("hdist", uint64(t.hdist)), log.Err(err))
+	}
+
+	// if genesis+1
+	if id == config.Genesis+1 {
+		return append(votes, mesh.GenesisBlock.Id()), nil
+	}
+
+	// not genesis, get from hare
+	bottom, top := calcHdistRange(id, t.hdist)
+
+	if res, err := t.hareResult.GetResult(bottom); err != nil { // no result for bottom, take the whole layer
+		t.With().Info("could not get result for bottom layer. adding the whole layer instead", log.Err(err),
+			log.Uint64("bottom", uint64(bottom)), log.Uint64("top", uint64(top)), log.Uint64("hdist", uint64(t.hdist)))
+		l, e := t.meshProvider.GetLayer(bottom)
```

Comment:
can replace with Get Layer IDs, you dont need entire block here

Reply:
Where is this function?

Reply:
Please add TODO here: we should pick the blocks in a smarter way than taking the entire layer.

Reply:
LayerBlockIds(index types.LayerID) ([]types.BlockID, error)

Fixed